### ISSUE TITLE
how to use fish as login shell

### ISSUE THREAD

**Issue by amonull** [tvrzna/emptty#81]
OS: void linux
shell: fish

when i try to use fish as my login shell in my ~/.emptty file as LoginShell=/bin/fish --login  i get the error:
ERROR: Xorg session finished with error, please check logs
the logs look like
![image](https://user-images.githubusercontent.com/90932550/210121377-04899248-5505-4bc2-9fa9-f4bf1ffa9e79.png)
i do not have this problem when using LoginShell=/bin/bash --login and my ~/.emptty file does have execute premissions.

**Comment by amonull** [comment]
i forgot to mention i used the command chsh -s /bin/fish to change from bash to fish

**Comment by amonull** [comment]
i've just tried using LoginShell=/bin/dash -l as well to see if this issue occurs in any other shell but it didn't. I think this is only an issue with fish.
And also logging in from a tty instead of emptty makes fish a login shell in that tty.

**Comment by amonull** [comment]
i noticed i forgot to add SESSION_ERROR_LOGGING info:
![image](https://user-images.githubusercontent.com/90932550/210124997-2f47e475-9c56-45bf-a8d0-1d3d90b654a5.png)
i think the problem comes from fish not being fully posix compliant so and even though i am using a shebang at the start of my ~/.emptty file i think that will need to be changed to a fish shebang and the script will need to be translated to fish from posix compliant shell for it to work.
i will translate my ~/.emptty to fish and try again to see if it works.

**Comment by amonull** [comment]
i translated the ~/.emptty script but ran into some issues.
After entering password you need to use $argv which i couldn't get working and using $argv[1] resulted in the error [1] not being defined being returned so at the end of the script instead of calling ```exec dbus-launch $argv/$argv[1]``` i just ran ```exec dbus-launch bspwm``` which allowed me to login but fish was still not a login shell and i don't think it processed any of the variables that are defined in the sh script as it should and may have just ignored all of it.
I don't know how this issue could be solved since even though a sh shebang is being used at the start of the .emptty script if LoginShell=/bin/fish --login is used the script is ran as a fish shell and not a sh/bash script.

~/.emptty as sh
```
#!/bin/sh
Selection=true
exec dbus-launch $@
```
~/.emptty as fish
```
#!/bin/fish
set Selection true
set LoginShell /bin/fish --login
exec dbus-launch bspwm    # using $argv or $argv[1] simply did not work
```

in the fish version i've also attempted to use a sh shebang at the start but use the fish syntax but i had the same error i described above that it ran but it simple didn't process any of the 

TLDR:
using `LoginShell=/bin/fish --login` causes ~/.emptty script to be interpreted as a fish script even if a sh/bash shebang is being used and script gives syntax errors due to fish incompatibility and if fish syntax is used variables like LoginShell do not get used as they are supposed to (i think default values get passed instead of the set values).

**Comment by tvrzna** [comment]
I'm not familiar with `fish`, however you can use `~/.emptty` file with `sh` and then set LoginShell to `fish`. Properties in that file are read per lines, so fish's syntax wouldn't pass the values into emptty. I'd recommend to try the standard `sh` shebang:
```
#!/bin/sh
Selection=true
LoginShell=/bin/fish --login

exec dbus-launch $@
```

I'm affraid `fish` as shebang is not compatible with `~/.emptty` file.

**Comment by amonull** [comment]
hi that is what i tried to do as well however the shebang gets ignored when `LoginShell=/bin/fish --login` is used and the script gets treated like a fish script.
The error from the session-errors.7.log.old is from using #!/bin/sh and LoginShell=/bin/fish --login at the same time.

**Comment by amonull** [comment]
i think what emptty is doing is when LoginShell is set to something it executes the .emptty file as the shell that LoginShell is set to regardless of the shebang

**Comment by tvrzna** [comment]
You are right, the shebang is ignored. It needs to get the fish compatibility.

**Comment by tvrzna** [comment]
Please, try latest version, if you are able to. There is simple fish support, if fish is defined in shebang. See [samples](https://github.com/tvrzna/emptty/blob/master/SAMPLES.md#script-with-fish-support-in-loginshell) for more informations.

**Comment by amonull** [comment]
thanks for the update just downloaded/built the latest version from github.
used the syntax provided from both the sample and my own config script using if statements both worked without any issues and after checking the log files there was nothing out of the ordinary, however after testing if the shell was login shell or not the result i got back was that it wasnt a login shell (tested for both fish and bash):

fish:
![image](https://user-images.githubusercontent.com/90932550/210299215-a0196081-7c32-42dc-8bdd-ccead823cb64.png)
bash:
![image](https://user-images.githubusercontent.com/90932550/210299246-d9eb8710-13a4-4fe1-8d4e-3f6e0c2564fb.png)

but after reading around i found out that this is just how xorg works and shells are not actually supposed to return is login shell if not logged in from the tty1-6.

thank you for fixing the issue.

**Comment by amonull** [comment]
it also might've returned false on checking if it was a login shell since i was checking from a terminal emulator either way i do think that was the expected and needed result and this issue can now be closed.
